### PR TITLE
Answer viewer fixes

### PR DESCRIPTION
**Pull request by maximusunc**
- Add pruning to the kg bubble graph
- Add pruning to the answer explorer graph
- Handle null kg and results in messages
- Update how we pick node category colors

### CODE REVIEW

File: frontend/src/utils/colors.js [hunk 1]
```diff
@@ -15,7 +15,6 @@ const conceptColorMap = {
   'biolink:Disease': '#fbb4ae', // Red
   'biolink:DiseaseOrPhenotypicFeature': '#fbb4ae', // Red, same as disease
   'biolink:Drug': '#8787ff', // Purply blue, same as chemical_substance
-  'biolink:Protein': '#ccebc5', // Green like gene
   'biolink:EnvironmentalFeature': '#8a9a5b', // Moss Green
   'biolink:Food': '#ffa343', // Neon Carrot
```
Comment (patrickkwang) [inline frontend/src/utils/colors.js:19]: Good band name.

**Comment by sarah-seitanakis** [comment]
image
The color of the middle node in the query graph and the table not matching the bubble graph and answer explorer graph is kind of confusing. I do see that the answer I selected has a middle node with category "small molecule". Since this is a sub category of the broader "molecular entity" can we make it a hue of that color? Or is there another way to make this less confusing?

**Comment by sarah-seitanakis** [comment]
image
Is there a way that we not have a pruning slider on graphs that will not require pruning? I don't like that these graphs have the pruning slider since a three node answer will not need it, nor will a bubble graph like this.

**Comment by sarah-seitanakis** [comment]
image
It looks like this answer has a publication but it is not showing in the answer explorer

**Review by sarah-seitanakis — CHANGES_REQUESTED** [review]
There are a couple things that I commented about

**Comment by maximusunc** [comment]
@sarah-seitanakis
- I updated the color mappings. Now we aren't specifying a color for small molecule, so nodes will go with the color of their closest parent category's color.
- I removed the prune slider for smaller graphs.
- The edge publication attribute format is not standardized, so I'm seeing new formats occasionally. I've updated the publication logic to include all the formats I'm aware of currently.

**Review by sarah-seitanakis — CHANGES_REQUESTED** [review]
image
Thanks for the fixes, it's looking great! One more thing-- do you think you could add some padding to the text that I highlighted in the answer explorer? Also, for some reason, the publication is still not showing up

**Comment by maximusunc** [comment]
@sarah-seitanakis That's a publication on a specific node, which I don't think we've ever handled? We've only ever showed publications for edges. @patrickkwang should we be showing links for these node publications?

**Comment by patrickkwang** [comment]
> should we be showing links for these node publications?

No, I'm not sure what those would be for. Unless someone asks for them, let's ignore.

**Review by sarah-seitanakis — APPROVED** [review]
Looks great!!

**Review by sarah-seitanakis — APPROVED** [review]

**Review by patrickkwang — APPROVED** [review]
Looks great!

**Pull request merged**